addons/mail/static/src/models/messaging_notification_handler.js

```diff
@@ -149,6 +151,20 @@ registerModel({
          * @param {Object} message
          */
         _handleNotification(message) {},
+        /**
+         * @private
+         * @param {Object} payload
+         * @param {Object[]} [payload.partners]
+         * @param {Object[]|undefined} [payload.guests]
+         */
+        _handleNotificationBusImStatus({ partners, guests }) {
+            if (partners) {
+                this.models['Partner'].insert(partners);
+            }
+            if (guests) {
+                this.models['Guest'].insert(guests);
+            }
```

Comment:
A bit sad that the field `im_status` in JS modeling cannot be easily changed, due to being heavily coupled with `im_status` in python :(